photon-test/src/main/scala/com/linkedin/photon/ml/test/TestTemplateWithTmpDir.scala

```diff
@@ -24,15 +27,25 @@ import org.testng.annotations.AfterMethod
  * @author dpeng
  */
 trait TestTemplateWithTmpDir {
+
+  private val tmpDirs = new mutable.ArrayBuffer[String]()
+
   /**
    * Return the temporary directory as a string.
    * @return the temporary directory as a string.
    */
-  def getTmpDir: String = TestTemplateWithTmpDir._tmpDirThreadLocal.get()
+  def getTmpDir: String = {
+    val tmpDirName = TestTemplateWithTmpDir._tmpDirThreadLocal.get()
+    tmpDirs.synchronized {
+      tmpDirs += tmpDirName
+    }
+    tmpDirName
+  }
 
   @AfterMethod
```

Comment:
@TDDFT Good to know, thanks!
@joshvfleming As mentioned in Issue #35, the problem we would like to address in this PR is that the afterMethod was not working properly. But FileUtils.forceDeleteOnExit is actually working fine, although it provides same functionality as the AfterClass method. So we may want to keep either the AfterClass method, or the FileUtils.forceDeleteOnExit method, but maybe not both.